src/main/groovy/com/lesfurets/jenkins/unit/PipelineTestHelper.groovy

```diff
@@ -687,7 +687,7 @@ class PipelineTestHelper {
             println stdout
         }
 
-        if (returnStdout) {
+        if (returnStdout && exitValue == 0) {
```

Comment:
FYI, you may want to `println stdout` when the error is raised, this is a behavior of regular `sh` script.

I.e. this test script
```
stdoutStr = sh(
  script:"""
    echo '123'
    false
  """,
  returnStdout: true
)
```
produces this output
```
[Pipeline] sh
11:36:59  + echo 123
11:36:59  + false
11:36:59  123
[Pipeline] }
[Pipeline] // script
[Pipeline] }
[Pipeline] // script
[Pipeline] }
[Pipeline] // stage
[Pipeline] stage
[Pipeline] { (Declarative: Post Actions)
[Pipeline] sh
11:37:08  + echo Will run cleanup
11:37:08  Will run cleanup
[Pipeline] }
[Pipeline] // stage
[Pipeline] }
[Pipeline] // timestamps
[Pipeline] }
[Pipeline] // timeout
[Pipeline] }
[Pipeline] // ansiColor
[Pipeline] }
[Pipeline] // node
[Pipeline] End of Pipeline
ERROR: script returned exit code 1
Finished: FAILURE
```

It's not a strong requirement, as this does not really affect testing.  I will leave it to your discretion

Reply:
Sure, that works for me. See 506173cb72f5664bbf99af3ea07c9a082a68b4d0.